src/OpenLoco/Math/Vector.hpp

```diff
@@ -0,0 +1,293 @@
+            if constexpr (TResolution > TVector2<T, TResolution2>::Resolution)
+            {
+                Base::x /= TResolution;
+                Base::y /= TResolution;
+                z /= TResolution;
+            }
+        }
+
+        constexpr bool operator==(const TVector3& rhs) const
+        {
+            return Base::operator==(rhs) && z == rhs.z;
+        }
+
+        constexpr bool operator!=(const TVector3& rhs) const
+        {
+            return !(*this == rhs);
+        }
+
+        constexpr TVector3& operator+=(const TVector3& rhs)
+        {
+            Base::operator+=(rhs);
+            z += rhs.z;
+            return *this;
+        }
+
+        constexpr TVector3& operator-=(const TVector3& rhs)
+        {
+            Base::operator-=(rhs);
+            z -= rhs.z;
+            return *this;
+        }
+
+        constexpr TVector3& operator*=(const T rhs)
+        {
+            Base::operator*=(rhs);
+            z *= rhs.z;
+            return *this;
+        }
+
+        constexpr TVector3& operator/=(const T rhs)
+        {
+            Base::operator/=(rhs);
+            z /= rhs.z;
+            return *this;
+        }
+
+        constexpr bool operator>=(const TVector3& rhs) const
+        {
+            return Base::operator>=(rhs) && z >= rhs.z;
+        }
+
+        constexpr bool operator<=(const TVector3& rhs) const
+        {
+            return Base::operator<=(rhs) && z <= rhs.z;
+        }
+
+        constexpr const TVector3 operator+(const TVector3& rhs) const
+        {
+            return { Base::x + rhs.x, Base::y + rhs.y, z + rhs.z };
+        }
+
+        constexpr const TVector3 operator-(const TVector3& rhs) const
+        {
+            return { Base::x - rhs.x, Base::y - rhs.y, z - rhs.z };
+        }
+
+        constexpr const TVector3 operator*(const T rhs) const
+        {
+            return { Base::x * rhs, Base::y * rhs, z * rhs };
+        }
+
+        constexpr const TVector3 operator/(const T rhs) const
+        {
+            return { Base::x / rhs, Base::y / rhs, z / rhs };
+        }
+    };
+#pragma pack(pop)
+
+    template<typename T, T TResolution>
+    static constexpr auto Rotate(const TVector2<T, TResolution>& vec, int32_t direction)
+    {
+        TVector2<T, TResolution> res;
+        switch (direction & 3)
+        {
+            default:
+            case 0:
+                res.x = vec.x;
+                res.y = vec.y;
+                break;
+            case 1:
+                res.x = vec.y;
+                res.y = -vec.x;
+                break;
+            case 2:
+                res.x = -vec.x;
+                res.y = -vec.y;
+                break;
+            case 3:
+                res.x = -vec.y;
+                res.y = vec.x;
+                break;
+        }
+        return res;
+    }
+
+    template<typename T, T TResolution>
+    static constexpr auto Distance(const TVector2<T, TResolution>& lhs, const TVector2<T, TResolution>& rhs)
```

Comment:
ManhattanDistance

We don't want to get this confused with computing the magnitude between two points which there is also equations for.

Reply:
Good point